src/widget/image.rs

```diff
@@ -0,0 +1,34 @@
+use super::{IntrinsicSize, Widget};
+use embedded_graphics::{image, prelude::*, primitives::Rectangle};
+
+#[derive(Clone, Copy)]
+pub struct Image<'a, T>
+where
+    T: ImageDrawable,
+{
+    image: &'a T,
+}
+
+impl<'a, T> Image<'a, T>
+where
+    T: ImageDrawable,
+{
+    pub fn new(image: &'a T) -> Self {
+        Self { image }
+    }
+}
+
+impl<'a, T, Display> Widget<Display> for Image<'a, T>
+where
+    T: 'a + ImageDrawable,
+    Display: DrawTarget<Color = T::Color>,
```

Comment:
So `Color` is like a "display-colors type" (eg `Rgb888`) and this is saying that the display has to have the same one of those as the image?

Reply:
Yeah, exactly.